File: source/clibCore/addonCollection_core/processAddonStyleEditor.vb

Option Explicit On
Option Strict On

Imports Contensive.BaseClasses
Imports Contensive.Core
Imports Contensive.Core.Models.Entity
Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
'
Namespace Contensive.Addons.Core
    Public Class processAddonStyleEditorClass
        Inherits Contensive.BaseClasses.AddonBaseClass
        '
        '====================================================================================================
        ''' <summary>
        ''' getFieldEditorPreference remote method
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Overrides Function execute(cp As Contensive.BaseClasses.CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim processor As CPClass = DirectCast(cp, CPClass)
                Dim cpCore As coreClass = processor.core
                '
                ' save custom styles
                If cpCore.authContext.isAuthenticated() And cpCore.authContext.isAuthenticatedAdmin(cpCore) Then
                    Dim addonId As Integer
                    addonId = cpCore.docProperties.getInteger("AddonID")
                    If (addonId > 0) Then
                        Dim styleAddon As Models.Entity.addonModel = Models.Entity.addonModel.create(cpCore, addonId)
                        If (styleAddon.StylesFilename.content <> cpCore.docProperties.getText("CustomStyles")) Then
                            styleAddon.StylesFilename.content = cpCore.docProperties.getText("CustomStyles")
                            styleAddon.save(cpCore)
                            '
                            ' Clear Caches
                            '
                            Call cpCore.cache.invalidateObject_Content(addonModel.contentName)
                        End If
                    End If
                End If
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: source/core/Addons/Primitives/processAddonStyleEditor.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Processor;
using Contensive.Processor.Models.DbModels;
using Contensive.Processor.Controllers;
using static Contensive.Processor.Controllers.genericController;
using static Contensive.Processor.constants;
//
namespace Contensive.Addons.Primitives {
    public class processAddonStyleEditorClass : Contensive.BaseClasses.AddonBaseClass {
        //
        //====================================================================================================
        /// <summary>
        /// getFieldEditorPreference remote method
        /// </summary>
        /// <param name="cp"></param>
        /// <returns></returns>
        public override object Execute(Contensive.BaseClasses.CPBaseClass cp) {
            string result = "";
            try {
                coreController core = ((CPClass)cp).core;
                //
                // save custom styles
                if (core.session.isAuthenticated & core.session.isAuthenticatedAdmin(core)) {
                    int addonId = core.docProperties.getInteger("AddonID");
                    if (addonId > 0) {
                       addonModel styleAddon =addonModel.create(core, addonId);
                        if (styleAddon.StylesFilename.content != core.docProperties.getText("CustomStyles")) {
                            styleAddon.StylesFilename.content = core.docProperties.getText("CustomStyles");
                            styleAddon.save(core);
                            //
                            // Clear Caches
                            //
                            core.cache.invalidateAllInContent(addonModel.contentName);
                        }
                    }
                }
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            return result;
        }
    }
}
